### PR TITLE
refactor(stream): SourceExecutor minor refactor

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #14687** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14687?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #14560** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14560?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #14706** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #14172** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14172?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #14465** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #14171** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14171?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment).

File: src/stream/src/executor/source/state_table_handler.rs [hunk 1]
```diff
@@ -180,11 +185,7 @@ impl<S: StateStore> SourceStateTableHandler<S> {
         Ok(())
     }
 
-    /// This function provides the ability to persist the source state
-    /// and needs to be invoked by the ``SourceReader`` to call it,
-    /// and will return the error when the dependent ``StateStore`` handles the error.
-    /// The caller should ensure that the passed parameters are not empty.
-    pub async fn take_snapshot<SS>(&mut self, states: Vec<SS>) -> StreamExecutorResult<()>
```
Comment (xxchan) [inline src/stream/src/executor/source/state_table_handler.rs:187]: I think the comments are verbose (and a little outdated?). 

Function name `set_states` might be enough to describe what it does.

**Comment by BugenZhao** [comment]
> > [!WARNING]
> > <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> > <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>
> 
> Current dependencies on/for this PR:
> * `main`
>   * **PR #14687** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14687?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
>     * **PR #14560** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14560?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
>       * **PR #14706** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
>         * **PR #14172** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14172?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
>           * **PR #14465** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
>       * **PR #14171** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14171?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
> 
> This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment).

Wow, such a complicated topology. 😁

**Comment by stdrc** [comment]
> > > [!WARNING]
> > > This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack [on Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment-downstack-mergeability-warning).
> > > [Learn more](https://graphite.dev/docs/merge-pull-requests)
> > 
> > 
> > Current dependencies on/for this PR:
> > 
> > * `main`
> >   
> >   * **PR [refactor(frontend): improve readability of LogicalSource #14687](https://github.com/risingwavelabs/risingwave/pull/14687)** [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14687?utm_source=stack-comment-icon)
> >     
> >     * **PR [refactor: minor refactor on fragment graph building #14560](https://github.com/risingwavelabs/risingwave/pull/14560)** [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14560?utm_source=stack-comment-icon)
> >       
> >       * **PR [refactor(stream): SourceExecutor minor refactor #14706](https://github.com/risingwavelabs/risingwave/pull/14706)** [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment-icon)  👈
> >         
> >         * **PR [feat(stream): add kafka backill executor #14172](https://github.com/risingwavelabs/risingwave/pull/14172)** [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14172?utm_source=stack-comment-icon)
> >           
> >           * **PR [feat: add_kafka_backfill_frontend #14465](https://github.com/risingwavelabs/risingwave/pull/14465)** [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14465?utm_source=stack-comment-icon)
> >       * **PR [feat(meta): align splits with source actor with an upstream #14171](https://github.com/risingwavelabs/risingwave/pull/14171)** [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14171?utm_source=stack-comment-icon)
> > 
> > This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14706?utm_source=stack-comment).
> 
> Wow, such a complicated topology. 😁

Just curious, can Graphite support inheriting from multiple upstream branches?

**Review by tabVersion — APPROVED** [review]

**Comment by xxchan** [comment]
@risingwavelabs/cargo-lock plz give an approve. Thanks

**Review by BugenZhao — APPROVED** [review]

**Review by stdrc — APPROVED** [review]

**Pull request merged**